hcloud/action_waiter.go

```diff
@@ -0,0 +1,102 @@
+package hcloud
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/samber/lo"
+)
+
+type ActionWaiter interface {
+	WaitForActionsFunc(ctx context.Context, actions []*Action, handleUpdate func(update *Action)) ([]*Action, error)
+	WaitForActions(ctx context.Context, actions []*Action) ([]*Action, error)
+	WaitForAction(ctx context.Context, action *Action) (*Action, error)
```

Comment:
> `actions []*Action` as argument is anoying. After creating a server I needed to contort myself to wait on all actions:
> ```
> WatchOverallProgress(ctx, append([]*Action{result.Action}, result.NextActions...)
> ```
> What do you think about accepting `actions ...*Action` instead?
> ```
> WaitForActions(ctx, result.Action, result.NextActions...)
> ```
> With the problem that you need to also do this if have just a slice (not sure we have ever encountered this in our integrations):
> ```
> WatchOverallProgress(ctx, result.NextActions)
> ```
> vs
> ```
> WaitForActions(ctx, result.NextActions...)
> ```

I already considered using variadic arguments for the actions, somehow didn't feel like it was the best API.

I think it depends on the intent behind having both `actions` and `next_actions`, if the first action fails, should we wait for the next actions ? How should the wait method handle a single action error, early exit, or wait until all actions are finished?

I believe most of the action waiting happens on a single action, so I would definitely want to simplify how we handle a single action. Having a few extra helpers for a list of actions seem sensible, but not a priority.

Using variadic arguments seem like a nice touch, I'll try to implement it, but we must answer the few questions above as well.

Reply:
The new smaller API implements variadic arguments for the actions.

Reply:
Go actually does not allow `WaitFor(ctx, result.Action, result.NextActions...)`, and you still need to use `appends()` for this. We have decided to keep it as is to make the case of a single action or a list of actions simpler. Only a small set of endpoints returns a single action + a list of actions.